src/test/regress/expected/mx_regular_user.out

```diff
@@ -25,18 +25,114 @@ SELECT start_metadata_sync_to_node('localhost', :worker_2_port);
+-- and GRANT access to the user
+CREATE SCHEMA "Mx Super User";
+SELECT 1 FROM run_command_on_workers($$CREATE SCHEMA "Mx Super User";$$);
+ ?column?
+---------------------------------------------------------------------
+        1
+        1
+(2 rows)
+
+SET citus.next_shard_id TO 2980000;
+SET search_path TO "Mx Super User";
+CREATE TABLE super_user_owned_regular_user_granted (a int PRIMARY KEY);
+SELECT create_reference_table ('"Mx Super User".super_user_owned_regular_user_granted');
+ create_reference_table
+---------------------------------------------------------------------
+
+(1 row)
+
+-- show that this table is owned by super user
+SELECT
+	rolsuper
+FROM
+	pg_roles
+		WHERE oid
+			IN
+		(SELECT relowner FROM pg_class WHERE oid = '"Mx Super User".super_user_owned_regular_user_granted'::regclass);
+ rolsuper
+---------------------------------------------------------------------
+ t
+(1 row)
+
+-- make sure that granting produce the same output for both community and enterprise
+SET client_min_messages TO ERROR;
+GRANT USAGE ON SCHEMA "Mx Super User" TO regular_mx_user;
+GRANT INSERT ON TABLE super_user_owned_regular_user_granted TO regular_mx_user;
+SELECT 1 FROM run_command_on_workers($$GRANT USAGE ON SCHEMA "Mx Super User" TO regular_mx_user;$$);
+ ?column?
+---------------------------------------------------------------------
+        1
+        1
+(2 rows)
+
+SELECT 1 FROM run_command_on_workers($$GRANT INSERT ON TABLE "Mx Super User".super_user_owned_regular_user_granted TO regular_mx_user;$$);
+ ?column?
+---------------------------------------------------------------------
+        1
+        1
+(2 rows)
+
+SELECT 1 FROM run_command_on_placements('super_user_owned_regular_user_granted', $$GRANT INSERT ON TABLE %s TO regular_mx_user;$$);
+ ?column?
+---------------------------------------------------------------------
+        1
+        1
+        1
+(3 rows)
+
+-- now that the GRANT is given, the regular user should be able to
+-- INSERT into the table
+\c - regular_mx_user - :master_port
+SET search_path TO "Mx Super User";
+COPY super_user_owned_regular_user_granted FROM STDIN;
+-- however, this specific user doesn't have UPDATE/UPSERT/DELETE/TRUNCATE
+-- permission, so  should fail
+INSERT INTO super_user_owned_regular_user_granted VALUES (1), (2) ON CONFLICT (a) DO NOTHING;
+ERROR:  permission denied for table super_user_owned_regular_user_granted
+TRUNCATE super_user_owned_regular_user_granted;
+ERROR:  permission denied for table super_user_owned_regular_user_granted
+CONTEXT:  while executing command on localhost:xxxxx
+DELETE FROM super_user_owned_regular_user_granted;
+ERROR:  permission denied for table super_user_owned_regular_user_granted
+UPDATE super_user_owned_regular_user_granted SET a = 1;
+ERROR:  permission denied for table super_user_owned_regular_user_granted
+-- AccessExclusiveLock == 8 is strictly forbidden for any user
+SELECT lock_shard_resources(8, ARRAY[2980000]);
+ERROR:  unsupported lockmode 8
```

Comment:
This test is currently not testing what we want, but when fixing it, we should keep a similar test that has a non existent lock mode. To be sure we handle that case correctly too.

Reply:
Well, thanks for catching, I didn't realize.

I guess we cannot add such a test with the current code now. I think we already guard against allowed lock modes in `IntToLockMode` and I don't think that it is a good idea to change that function?

I'm really OK to keep things as is just in case someone changes `IntToLockMode`, we still have one extra tests that would remind the executor doesn't acquire `AccessExclusiveLock` (at the moment at least).

Reply:
Okay, I think it's fine to depend on `IntToLockMode`  for this. Let's keep this test and its output, but let's remove the code checking explicitly for AccessExclusiveLock (at my other comment).

I think it also still makes sense to add a test with a really non existent lock mode, to future proof the tests against someone changing IntToLockMode.

Reply:
> I think it also still makes sense to add a test with a really non existent lock mode, to future proof the tests against someone changing IntToLockMode.

That test already exists in another file